### DISCUSSION TITLE
Personal Access Token; Why isn't there a Repo:Read scope?

### DISCUSSION THREAD

**Discussion by james-perfectserve** [community/community#33057] [Code Security]
I need a token that will allow read access to a private repo, but **I don't want it to have Write access.**
I've looked through the list of available scopes multiple times, I thought it might be read:project but project is a diff thing than a repo.
Is there seriously no way to generate a read only Personally Access Token?

https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes

**Comment by HariSekhon** [comment]
This is a common long-running request, wish GitHub would pick this up:

#5317

#18666
